lib/posix/pthread.c

```diff
@@ -143,6 +142,31 @@ static void zephyr_thread_wrapper(void *arg1, void *arg2, void *arg3)
 	pthread_exit(NULL);
 }
 
+static bool pthread_attr_is_valid(const struct pthread_attr *attr)
+{
+	/*
+	 * FIXME: Pthread attribute must be non-null and it provides stack
```

Comment:
Hrm. This seems a bit unfortunate. Could the stack be dynamically allocated from the user heap if a user heap were available?

Reply:
@keith-packard please note that we already has this restriction (rather than introducing it in this PR) - so it shouldn't be fixed by this PR.

Reply:
> Hrm. This seems a bit unfortunate. Could the stack be dynamically allocated from the user heap if a user heap were available?

That is the topic of a longstanding issue & series of PRs I've been working on (in my "spare time") for... "a while" 😅

Truthfully, this should be handled at the Zephyr level and not at the pthreads level.

Currently blocked on MMU issues.